src/Features/Core/Portable/Completion/CompletionService.ProviderManager.cs

```diff
@@ -234,8 +263,16 @@ public TestAccessor(ProviderManager providerManager)
                 public ImmutableArray<CompletionProvider> GetProviders(ImmutableHashSet<string> roles, Project? project)
                 {
                     using var _ = ArrayBuilder<CompletionProvider>.GetInstance(out var providers);
+
                     providers.AddRange(_providerManager.GetImportedAndBuiltInProviders(roles));
-                    providers.AddRange(GetProjectCompletionProviders(project));
+
+                    if (project != null)
+                    {
+                        _providerManager._projectProvidersWorkQueue.AddWork(project.AnalyzerReferences);
+                        _providerManager._projectProvidersWorkQueue.WaitUntilCurrentBatchCompletesAsync().Wait();
```

Comment:
Can we just make this test accessor method async?

Reply:
Done